### DISCUSSION TITLE
Tags Index: Option to render URL text from Markdown Headers instead of page.title

### DISCUSSION THREAD

**Discussion by anna-engle** [squidfunk/mkdocs-material#5629] [Ideas]
Currently, the [`tags`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-tags/#adding-a-tags-index) plugin allows for a [Tags Index](https://squidfunk.github.io/mkdocs-material/setup/setting-up-tags/#+tags.tags_file) to be rendered via the `tags_file` definition. 

![tags-index](https://github.com/squidfunk/mkdocs-material/assets/22220925/43472531-248e-408d-9aca-83e1798ed501)

In the Index, the URL text is generated from the `page.title`, either explicitly defined with [front matter](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-title):

##### welcome-intro.md
``` 
--- 
tags:
   - Introduction
   
title: "This is my page title"
---

Some text
```

or, it is generally rendered from the title listed in the nav. For example, consider the following snippet:

##### mkdocs.yml
```
nav:
  - "Welcome":
      - "Overview": welcome-intro.md
```

In the above example, assuming `page.title` is **not** explicitly set, the URL text in the index would render as `Overview`.

This can be a problem for users in a few different cases. Mainly, say we have the following two markdown files:

##### welcome-intro.md
``` 
--- 
tags:
   - Introduction
---

# Welcome: an Introduction
```

##### debugging-intro.md
``` 
--- 
tags:
   - Introduction
---

# Debugging: an Introduction
```

And the following nav defined:

##### mkdocs.yml
```
nav:
  - "Welcome":
      - "Overview": welcome-intro.md
  - "Debugging":
      - "Overview": debugging-intro.md
```

In this case, the Tags Index would render something like:

```
[Introduction]
- Overview
- Overview
```

This is not very intuitive to users from this viewpoint, if two pages have the same page title. The workaround would be to set the `page.title` explicitly in order to render a more intuitive link, however this could be cumbersome, especially if you have a large set of tagged files like this.

It would be nice to give authors the option to render the URL text from the Markdown header itself, something like:

```
  tags:
    tags_file: tags.md
       use_markdown_header: true
```

This way, the Tags Index in the above example would render:

```
[Introduction]
- Welcome: an Introduction
- Debugging: an Introduction
```

**Comment by squidfunk** [comment]
It is the exact page title that MkDocs [determines for the page](https://github.com/mkdocs/mkdocs/blob/66f8df97d282f819344f245e42ee1b8100e1bb24/mkdocs/structure/pages.py#L225-L248), since for rendering the tags, we use `page.title`:

https://github.com/squidfunk/mkdocs-material/blob/9e0c005396ddb3a7534c9906c2b3d0d3c9476ab9/src/plugins/tags/plugin.py#L148-L150

We're planning to do a refactoring of the tags plugin in the near future, so if you want, you can raise a change request and lay out your use case(s). The more input we have, the better we can gravitate towards a better solution 😊 We actually already have a (yet unshared) prototype that is much more powerful, so any input is welcome.
